types/report.go

```diff
@@ -37,3 +51,40 @@ type Error struct {
 	Label   string
 	Message string
 }
+
+func (c *CoverageReport) AddCoverageFromState(resourceId, resourceType, address string, jsonBody map[string]interface{}) error {
```

Comment:
>AddCoverageFromState [](http://example.com/codeflow?start=25&length=20)

could it be moved into `coverage` package?

Reply:
should this be better to stay with the `CoverageReport` struct

Reply:
I think the original design logic is to put all struct definitions in the `package types`. But the methods should not be defined here. If you consider defining it as a member method of CoverageReport, both it and `CoverageReport` should be defined in a new file. If you consider defining it as a global function, it should be put in the `package tf`

Reply:
Thanks, this makes sense. I will change the code.